core/src/apps/management/apply_settings.py

```diff
@@ -11,6 +11,16 @@
     from trezor.messages.ApplySettings import ApplySettings
 
 
+def humanize_time(milliseconds: int) -> str:
+    units = (("hour", 60 * 60 * 1000), ("minute", 60 * 1000), ("second", 1000))
+    for (u, divisor) in units:
+        if milliseconds >= divisor:
+            num = milliseconds // divisor
+            return "{} {}{}".format(num, u, "s" if num > 1 else "")
+
+    return "{} ms".format(milliseconds)
```

Comment:
Maybe this function could benefit of having a unit test? These are in `core/tests`.

Reply:
Done 8516e7a.